common/lib/xmodule/xmodule/tests/test_capa_module.py

```diff
@@ -3236,3 +3236,16 @@ def test_generate_report_data_skip_dynamath(self):
         iterator = iter([self._user_state(suffix='_dynamath')])
         report_data = list(descriptor.generate_report_data(iterator))
         assert 0 == len(report_data)
+
+    def test_generate_report_data_report_loncapa_error(self):
+        #Test to make sure reports continue despite loncappa errors, and write them into the report.
+        descriptor = self._get_descriptor()
+        with patch('xmodule.capa_module.LoncapaProblem') as mock_LoncapaProblem:
+            mock_LoncapaProblem.side_effect = LoncapaProblemError
```

Comment:
Nit: An iterable can be passed into `side_effect` and simulate a single exception along with other successes:
https://docs.python.org/3.8/library/unittest.mock-examples.html#side-effect-functions-and-iterables
Then you could test for both successes and failures below - which would be a closer test to the actual issue. But all failures seems fine as well.

Reply:
I'll keep this in mind for sure!